flask-server/main.py

```diff
@@ -338,6 +338,20 @@ def add_clink():
         }
         return response_object, 401
 
+@app.route('/apis/add-readmap/<string:user_id>', methods=['POST'])
+def add_readmap(user_id):
+    read_entity = datastore.Entity(key=datastore_client.key('user_read_map'))
+    mapping = {
+        'clink_id': int(request.json['clink']),
+        'user_id': int(user_id)
+    }
+    read_entity.update(mapping)
+    datastore_client.put(read_entity)
+
+    key = datastore_client.key('clink', int(request.json['clink']))
+    clink = list(datastore_client.query(kind='clink').add_filter('__key__', '=', key).fetch(limit=1))[0]
```

Comment:
`clink = datastore_client.get(key)` instead of fetching, it is the same thing just simplified